plugins/org.komodo.core/src/org/komodo/repository/RepositoryImpl.java

```diff
@@ -1243,4 +1254,18 @@ public void unpublish( final UnitOfWork uow,
 
     }
 
+    @Override
+    public KomodoObject komodoRoot(final UnitOfWork uow) throws KException {
+        return create(uow, KOMODO_ROOT, KomodoLexicon.Komodo.NODE_TYPE);
+    }
+
+    @Override
+    public KomodoObject komodoLibrary(final UnitOfWork uow) throws KException {
+        return create(uow, LIBRARY_ROOT, KomodoLexicon.Library.NODE_TYPE);
+    }
+
+    @Override
+    public KomodoObject komodoWorkspace(final UnitOfWork uow) throws KException {
```

Comment:
Sorted.
I have also removed the null return in getParent() if the parent of workspace is requested since, although it would be rarely required, it seems wrong to request a node's parent of a node that has a parent and get null.